Install: bring build doc up to date for style (#1503)
fluent/fluent-bit-docs: installation/sources/build-and-install.md

# Build and install

[Fluent Bit](http://fluentbit.io) uses [CMake](http://cmake.org) as its build system.

## Requirements

- CMake 3.12 or greater. You might need to use `cmake3` instead of `cmake`.
- Flex

> - Bison 3 or greater

Reviewer: 🦬 🦬 🦬

- YAML headers
- OpenSSL headers

## Prepare environment

> If you already know how CMake works you can skip this part and look at the build
> options available.

Reviewer: ```suggestion
If you already know how CMake works, you can skip this section and review the
available [build options](#general-options).
```

Reviewer: Suggest adding a nice little cross link.

Author: I'm going to worry about crosslinks in a later iteration :D

The following steps explain how to build and install the project with the default
options.

1. Change to the `build/` directory inside the Fluent Bit sources:

  ```bash
  cd build/
  ```

1. Let [CMake](http://cmake.org) configure the project specifying where the root
   path is located:

  ```bash
  cmake ../
  ```

>   This should display a series of results like:

Reviewer: ```suggestion
  This command displays a series of results similar to:
```

  ```text
  -- The C compiler identification is GNU 4.9.2
  -- Check for working C compiler: /usr/bin/cc
  -- Check for working C compiler: /usr/bin/cc -- works
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- The CXX compiler identification is GNU 4.9.2
  -- Check for working CXX compiler: /usr/bin/c++
  -- Check for working CXX compiler: /usr/bin/c++ -- works
  ...
  -- Could NOT find Doxygen (missing:  DOXYGEN_EXECUTABLE)
  -- Looking for accept4
  -- Looking for accept4 - not found
  -- Configuring done
  -- Generating done
  -- Build files have been written to: /home/edsiper/coding/fluent-bit/build
  ```

1. Start the compilation process using the `make` command:

  ```bash
  make
  ```

>   This should display results like:

Reviewer: ```suggestion
  This command displays results similar to:
```

  ```text
  Scanning dependencies of target msgpack
  [  2%] Building C object lib/msgpack-1.1.0/CMakeFiles/msgpack.dir/src/unpack.c.o
  [  4%] Building C object lib/msgpack-1.1.0/CMakeFiles/msgpack.dir/src/objectc.c.o
  [  7%] Building C object lib/msgpack-1.1.0/CMakeFiles/msgpack.dir/src/version.c.o
  ...
  [ 19%] Building C object lib/monkey/mk_core/CMakeFiles/mk_core.dir/mk_file.c.o
  [ 21%] Building C object lib/monkey/mk_core/CMakeFiles/mk_core.dir/mk_rconf.c.o
  [ 23%] Building C object lib/monkey/mk_core/CMakeFiles/mk_core.dir/mk_string.c.o
  ...
  Scanning dependencies of target fluent-bit-static
  [ 66%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_pack.c.o
  [ 69%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_input.c.o
  [ 71%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_output.c.o
  ...
  Linking C executable ../bin/fluent-bit
  [100%] Built target fluent-bit-bin
  ```

1. To continue installing the binary on the system, use `make install`:

```bash
make install
```

> You might need root privileges. You can try to prefix the command with `sudo`.

Reviewer: ```suggestion
If the command indicates insufficient permissions, prefix the command with `sudo`.
```

| `FLB_STREAM_PROCESSOR` | Enable Stream Processor | _Yes_ |
| `FLB_CONFIG_YAML` | Enable YAML configuration support | _Yes_ |
| `FLB_WASM` | Build with WASM runtime support | _Yes_ |
| `FLB_WAMRC` | Build with WASM AOT compiler executable | _No_ |

### Development options

| Option | Description | Default |
| :--- | :--- | :--- |
| `FLB_DEBUG` | Build binaries with debug symbols | _No_ |
| `FLB_VALGRIND` | Enable Valgrind support | _No_ |
| `FLB_TRACE` | Enable trace mode | _No_ |
| `FLB_SMALL` | Minimise binary size | _No_ |
| `FLB_TESTS_RUNTIME` | Enable runtime tests | _No_ |
| `FLB_TESTS_INTERNAL` | Enable internal tests | _No_ |
| `FLB_TESTS` | Enable tests | _No_ |
| `FLB_BACKTRACE` | Enable backtrace/stacktrace support | _Yes_ |

### Optimization options

| Option | Description | Default |
| :--- | :--- | :--- |
| `FLB_MSGPACK_TO_JSON_INIT_BUFFER_SIZE` | Determine initial buffer size for `msgpack` to `json` conversion in terms of memory used by payload. | `2.0` |
| `FLB_MSGPACK_TO_JSON_REALLOC_BUFFER_SIZE` | Determine percentage of reallocation size when `msgpack` to `json` conversion buffer runs out of memory. | `0.1` |

### Input plugins

> Input plugin gather information from a specific source type like network interfaces,

Reviewer: ```suggestion
Input plugins gather information from a specific source type like network interfaces,
```
